openerp/models.py

```diff
@@ -1865,14 +1865,20 @@ def _read_group_fill_results(self, cr, uid, domain, groupby, remaining_groupbys,
         result = []
         known_values = {}
         def append_left(left_side):
-            grouped_value = left_side[groupby] and left_side[groupby][0]
+            if isinstance(left_side[groupby], (list, tuple)):
```

Comment:
Without commenting the code itself, this kind of modification should totally be part of an independant commit and PR, and probably an explanation and maybe a test.